### PR TITLE
Restore ability to pickle dask arrays

### LINKED ISSUE #6589: Cannot pickle dask array objects
**What happened**:

Attempting to pickle a dask array gives:
```
AttributeError: Can't pickle local object 'broadcast_trick.<locals>.inner'
```

Dask arrays are no longer pickleable since v2.23.0.

**What you expected to happen**:

The dask array can be pickled.

**Minimal Complete Verifiable Example**:

```python
import dask.array
import pickle
a = dask.array.ones((10, 10))
pickle.dumps(a)
```

This fails on versions larger than v2.22.

**Anything else we need to know?**:

I am happy to fix this. It seems like a simple fix to move the `inner` function to the module level.

**Environment**:

- Dask version: 2.25.0
- Python version: 3.7.7
- Operating System: Mac OS X
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by nbren12**
Dask arrays outputted by `dask.array.ones` can no longer be pickled because of a change in how `broadcast_trick` is implemented. This went unnoticed by the existing pickle/unpickle test, which only tested pickling for the output of `da.arange`.  This PR adds a failing test case and repairs it.

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

Resolves #6589

### CODE REVIEW

File: dask/array/tests/test_array_core.py [hunk 1]
```diff
@@ -2413,12 +2413,7 @@ def test_from_array_scalar(type_):
 
     dx = da.from_array(x, chunks=-1)
     assert_eq(np.array(x), dx)
-    assert isinstance(
-        dx.dask[
-            dx.name,
-        ],
-        np.ndarray,
-    )
+    assert isinstance(dx.dask[dx.name,], np.ndarray,)
```
Comment (nbren12) [inline dask/array/tests/test_array_core.py:2416]: This is a whitespace change from `black`. What version does the CI use?

File: dask/array/wrap.py [hunk 2]
```diff
@@ -197,12 +201,7 @@ def full_like(a, fill_value, *args, **kwargs):
         raise ValueError(
             f"fill_value must be scalar. Received {type(fill_value).__name__} instead."
         )
-    return _full_like(
-        a=a,
-        fill_value=fill_value,
-        *args,
-        **kwargs,
-    )
+    return _full_like(a=a, fill_value=fill_value, *args, **kwargs,)
```
Comment (nbren12) [inline dask/array/wrap.py:204]: Similarly, this is a false positive black change.

**Review by TomAugspurger — APPROVED** [review]
The CI failure is being fixed in https://github.com/dask/dask/pull/6598. I'll merge master after that's in.

File: dask/array/tests/test_array_core.py (hunk 1 above)
Comment (jrbourbeau) [reply dask/array/tests/test_array_core.py:2416]: It should use the latest release of `black` (today that's `20.8b1`)

**Review by jrbourbeau — APPROVED** [review]
LGTM, thanks @nbren12!

**Pull request merged**